### PR TITLE
Update analyzers.md

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: docs/project/analyzers.md [hunk 1]
```diff
@@ -3,21 +3,15 @@
 This repo relies on [.NET Compiler Platform analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019) to help validate the correctness, performance, and maintainability of the code.  Several existing analyzer packages are wired into the build, but it is easy to augment the utilized packages in order to experiment with additional analyzers.
```
Comment (elachlan) [inline docs/project/analyzers.md:3]: Could you remove the localization `en-us` in the URL.

**Comment by elachlan** [comment]
Maybe we could add a section about https://learn.microsoft.com/dotnet/core/tools/dotnet-format? Which would be useful to apply an analyzer fix across the code base.

File: docs/project/analyzers.md (hunk 1 above)
Comment (Youssef1313) [inline docs/project/analyzers.md:3]:
```suggestion
This repo relies on [.NET Compiler Platform analyzers](https://learn.microsoft.com/visualstudio/code-quality/roslyn-analyzers-overview) to help validate the correctness, performance, and maintainability of the code.  Several existing analyzer packages are wired into the build, but it is easy to augment the utilized packages in order to experiment with additional analyzers.
```

**Comment by elachlan** [comment]
LGTM!!

File: docs/project/analyzers.md [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
-    <Rule Id="S1075" Action="None" />
-    ...
-  </Rules>
-```
+1. Select a package you want to employ, for example https://www.nuget.org/packages/SonarAnalyzer.CSharp/.  This analyzer package's name is `SonarAnalyzer.CSharp` and the latest version as of this edit is `8.50.0.58025`.
+2. Add a `PackageReference` entry to <https://github.com/dotnet/runtime/blob/main/eng/Analyzers.props>, e.g.
+    ```XML
+    <PackageReference Include="SonarAnalyzer.CSharp" Version="SonarAnalyzerCSharpVersion" PrivateAssets="all" />
+    ```
+3. Add `<SonarAnalyzerCSharpVersion>8.50.0.58025</SonarAnalyzerCSharpVersion>` property to [Versions.props](https://github.com/dotnet/runtime/blob/main/eng/Versions.props).
```
Comment (stephentoub) [inline docs/project/analyzers.md:11]: This should be a general statement about what's being done followed by an example, just as the previous two bullets.

That said, this splitting of the version into the separate Version.props file is really only necessary if we're actually adding the analyzer in for everyone.  If you're just adding in additional analyzers for testing locally, this is unnecessary overhead.

File: docs/project/analyzers.md [hunk 3]
```diff
@@ -1,27 +1,23 @@
...
+4. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  Rules can be disabled by adding entries to the globalconfig files [CodeAnalysis.src.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.src.globalconfig) or [CodeAnalysis.test.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.test.globalconfig).
```
Comment (stephentoub) [inline docs/project/analyzers.md:12]:
```suggestion
4. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  You can change the severity for rules by adding entries to the globalconfig files [CodeAnalysis.src.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.src.globalconfig) or [CodeAnalysis.test.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.test.globalconfig).
```

File: docs/project/analyzers.md [hunk 4]
```diff
@@ -1,27 +1,23 @@
...
+The build system in this repo defaults to treating all warnings as errors. It can be helpful when enabling a new rule to temporarily allow warnings to be warnings rather than errors, while you proceed to fix all of them across the repo. Instead of building from the root of the repo with:
 ```
 build.cmd
 ```
 (or `./build.sh` on Unix), warnings-as-errors can be disabled with:
 ```
 build.cmd -warnAsError 0
 ```
+
+**Tip:** You can use [dotnet format](https://learn.microsoft.com/dotnet/core/tools/dotnet-format) to fix analyzer warnings if the analyzer has an associated `CodeFixProvider`.
```
Comment (stephentoub) [inline docs/project/analyzers.md:23]:
```suggestion
```

File: docs/project/analyzers.md [hunk 5]
```diff
@@ -1,23 +1,17 @@
...
-  <Rules AnalyzerId="SonarAnalyzer.CSharp" RuleNamespace="SonarAnalyzer.CSharp">
-    <Rule Id="S1006" Action="None" />
-    <Rule Id="S1075" Action="None" />
-    ...
-  </Rules>
-```
+1. Select a package you want to employ, for example https://www.nuget.org/packages/SonarAnalyzer.CSharp/.  This analyzer package's name is `SonarAnalyzer.CSharp` and the latest version as of this edit is `8.50.0.58025`.
+2. Add a `PackageReference` entry to <https://github.com/dotnet/runtime/blob/main/eng/Analyzers.props>, e.g.
+    ```XML
+    <PackageReference Include="SonarAnalyzer.CSharp" Version="SonarAnalyzerCSharpVersion" PrivateAssets="all" />
```
Comment (Youssef1313) [inline docs/project/analyzers.md:9]:
```suggestion
    <PackageReference Include="SonarAnalyzer.CSharp" Version="8.50.0.58025" PrivateAssets="all" />
```

File: docs/project/analyzers.md [hunk 6]
```diff
@@ -1,23 +1,17 @@
...
+    ```
+3. Add `<SonarAnalyzerCSharpVersion>8.50.0.58025</SonarAnalyzerCSharpVersion>` property to [Versions.props](https://github.com/dotnet/runtime/blob/main/eng/Versions.props).
```
Comment (Youssef1313) [inline docs/project/analyzers.md:11]:
```suggestion
```

File: docs/project/analyzers.md [hunk 7]
```diff
@@ -1,23 +1,17 @@
...
+4. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  You can change the severity for rules by adding entries to the globalconfig files [CodeAnalysis.src.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.src.globalconfig) or [CodeAnalysis.test.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.test.globalconfig).
```
Comment (Youssef1313) [inline docs/project/analyzers.md:12]:
```suggestion
3. After that point, all builds will employ all rules in that analyzer package that are enabled by default.  You can change the severity for rules by adding entries to the globalconfig files [CodeAnalysis.src.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.src.globalconfig) or [CodeAnalysis.test.globalconfig](https://github.com/dotnet/runtime/blob/main/eng/CodeAnalysis.test.globalconfig).
```

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**